### DISCUSSION TITLE
使用最新版编译出linux libpag.a编译ffmpeg时报错undefined reference

### DISCUSSION THREAD

**Discussion by wangyinlong** [Tencent/libpag#2297] [Q&A] [answered]
我首先在linux目录下./build_pag.sh编译得到了include头文件和libpag.a静态库
然后使用include和libpag.a重新编译libavfilter/cpp，将静态库libpag.a生成pag_impl.a

然后重新编译ffmpeg
./configure --prefix=/usr --enable-static --enable-shared --enable-pic --enable-pthreads --enable-version3 --enable-gnutls --enable-gpl \
    --enable-nonfree --enable-libass --enable-libfreetype --enable-libmp3lame --enable-libtheora --enable-libvorbis --enable-libx264 \
    --disable-filter=gltransition --enable-filter=pag \
    --extra-cflags="-I/usr/local/include/ -I/data/home/johnnylwang/gffmpeg_johnny1/third_party/swiftshader/include" \
    --extra-ldflags="-L/usr/local/lib -L/data/home/johnnylwang/gffmpeg_johnny1/third_party/swiftshader/linux -L/data/home/johnnylwang/gffmpeg_johnny1/G-ffmpeg/libavfilter/cpp" \
    --extra-libs="-lpag_impl -lEGL -lGLESv2  -lstdc++ -lz -ldl -lpthread -lm"

make

在make的时候遇到了如下报错
<img width="1642" alt="image" src="https://github.com/Tencent/libpag/assets/5877234/41ae0d2b-9912-479c-924d-25f64e254d7f">
请问是这些函数的实现已经在新版的代码中去除掉了吗？还是什么原因呀？

**Comment by kevingpqi123** [comment]
pag 的每一次提交都会触发 linux 构建，编译通过后才能合入
可以看下这个文件中 linux 编译的部分：
https://github.com/Tencent/libpag/blob/main/.github/workflows/build.yml

整个编译过程这里都可以看得到
https://github.com/Tencent/libpag/actions/runs/8964994047/job/24617672845

**Comment by domchen** [comment, marked as answer]
你这个pag_impl.a以及  --enable-filter=pag 是自己写了个ffmpeg的pag扩展吗？这个报错和pag库应该没关系，看内容是你忘记了链接libpag.a的静态库或者链接的顺序错了没找到定义。
